### PR TITLE
Add tests for tax display for the shopper

### PR DESCRIPTION
**Pull request by lanej0**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR adds a number of end-to-end tests for checking the tax display settings in the store, cart and checkout.

Closes https://github.com/woocommerce/woocommerce-quality/issues/701

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure tests pass in CI
2. Pull this branch
3. `pnpm env:start && pnpm test:e2e-pw shopper/checkout-calculate-tax.spec.js`
4. `pnpm env:restart && pnpm test:e2e-pw`

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Comment by veljkho** [comment]
Hi @lanej0 
Could you rename your test to `checkout-calculate-tax.spec.js`?
I will have the same tax test for `checkout-block` where I will name it `checkout-block-calculate-tax.spec.js`.

**Comment by lanej0** [comment]
No problem @veljkho -- renamed!

File: plugins/woocommerce/tests/e2e-pw/tests/shopper/checkout-calculate-tax.spec.js [hunk 1]
```diff
@@ -0,0 +1,692 @@
...
+
+test.describe( 'Shopper Tax Display Tests', () => {
+
+	test.beforeAll( async ( { baseURL } ) => {
+		const api = new wcApi( {
+			url: baseURL,
+			consumerKey: process.env.CONSUMER_KEY,
+			consumerSecret: process.env.CONSUMER_SECRET,
+			version: 'wc/v3',
+		} );
```
Comment (alopezari) [inline plugins/woocommerce/tests/e2e-pw/tests/shopper/checkout-calculate-tax.spec.js:20]: Hmm not something for this PR but I think we would make it easier if we just defined a public general object:
```
v3: {
		url: baseURL,
		consumerKey: process.env.CONSUMER_KEY,
		consumerSecret: process.env.CONSUMER_SECRET,
		version: 'wc/v3',
}
```

And then we just initialize the API call as `const api = new wcApi( v3 );`

This way we would substantially reduce code duplication on API calls, where we often (or maybe always) use the same attribute values.

Comment (nigeljamesstevenson) [reply plugins/woocommerce/tests/e2e-pw/tests/shopper/checkout-calculate-tax.spec.js:20]: Great idea. I have created a [task](https://github.com/woocommerce/woocommerce-quality/issues/738) for this, although, https://github.com/woocommerce/woocommerce-quality/issues/737 is worth bearing in mind.

Comment (nigeljamesstevenson) [reply plugins/woocommerce/tests/e2e-pw/tests/shopper/checkout-calculate-tax.spec.js:20]: I updated the testing instructions to use pnpm test:e2e-pw shopper/checkout-calculate-tax.spec.js

**Comment by nigeljamesstevenson** [comment]
For this one I am seeing a failing test when executing `pnpm env:start && pnpm test:e2e-pw shopper/checkout-calculate-tax.spec.js`

`shopper/checkout-calculate-tax.spec.js:81:2 › Shopper Tax Display Tests › checks that taxes are calculated properly on totals, inclusive tax displayed properly` 

image

This occurs on a newly started environment as we see the following:
image

**Comment by lanej0** [comment]
Thanks @nigeljamesstevenson -- I had some logic problems in my test resetting.  It was passing on the first run through, but wouldn't on subsequent runs (without resetting the environment).  I've fixed that now!

**Review by nigeljamesstevenson — APPROVED** [review]
Testing great now - @lanej0 - thanks!

**Comment by nigeljamesstevenson** [comment]
I see some jobs had failed so have kicked them off again in case it was due to flakiness:

- Run tests against PR / Runs E2E tests
- Run tests against PR in an environment with HPOS disabled / Runs E2E tests with HPOS disabled

**Comment by nigeljamesstevenson** [comment]
I see a failure on your new test in [one of the jobs](https://github.com/woocommerce/woocommerce/actions/runs/6713307630/job/18260502480?pr=40178) that I was able to recreate locally when executing all the tests: 

image

Possibly with all tests running concurrently (or whatever environment state the previously executed tests set the environment in prior to the execution of the new test) this is causing issues. I ran this test individually after the initial failure and it passed so the post test cleanup is working.

**Pull request merged**